### PR TITLE
Minor updates to code

### PR DESCRIPTION
**Pull request by rbedfordpro**
Added a few purchases: Shattrath Portal & Reagent Vendor
Added new config and logic to allow giving other ranks ability to make purchases
Updated with missing Includes (per WiZZy)

### CODE REVIEW

**Comment by Stoabrogga** [comment]
According to @Knindzagxg there's also sometimes a problem concerning phasing, e.g. in Wintergrasp. I think the problem lies here:
https://github.com/azerothcore/mod-guildhouse/blob/b35e7067308f3a783488f0625efab6a733a85f8f/src/mod_guild_house_v2.cpp#L198-L211
This code only applies the phase of the first aura, but all auras have to be considered. I think it is better to replace the function with something like this:
```cpp
    uint32 GetNormalPhase(Player* player) const
    {
        if (player->IsGameMaster())
            return PHASEMASK_ANYWHERE;

        uint32 phase = player->GetPhaseByAuras();
        if (!phase)
            return PHASEMASK_NORMAL;
        else
            return phase;
    }
```
Perhaps this can be added to this PR.

File: sql/world/Creatures.sql [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
 (18649, 0, 0, 0, 0, 0, 18234, 0, 0, 0, 'Innkeeper Monica', NULL, NULL, 0, 1, 2, 0, 35, 65537, 0.8, 0.28571, 1, 0, 2, 2, 0, 26, 4.6, 2000, 1900, 1, 0, 2048, 0, 0, 0, 0, 0, 0, 1, 1, 100, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 1, 3, 1, 1, 1, 1, 0, 70, 1, 0, 2, 'npc_innkeeper', 12340),
 (33251, 0, 0, 0, 0, 0, 28516, 0, 0, 0, 'Death Knight Trainer', '', '', 9691, 80, 80, 2, 35, 51, 1, 1.14286, 1, 1, 422, 586, 0, 642, 7.5, 2000, 2000, 1, 32768, 2048, 0, 0, 0, 0, 6, 0, 345, 509, 103, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 0, 3, 1, 10, 1, 1, 0, 0, 1, 8388624, 0, '', 12340),
 (70102, 0, 0, 0, 0, 0, 25901, 0, 0, 0, 'Speedfang', 'Guild House Assistant', '', 0, 35, 35, 0, 35, 1, 1, 1.14286, 1, 0, 49, 64, 0, 118, 1, 2000, 2000, 1, 33536, 2048, 0, 0, 0, 0, 0, 0, 33, 49, 11, 7, 4096, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 0, 3, 1, 1, 1, 1, 0, 0, 1, 0, 0, 'GuildHouseSpawner', 12340);
+>>>>>>> 9ffd5adbbb2071bb539e752a15b4ecd3d62a9d7b
```
Comment (FrancescoBorzi) [inline sql/world/Creatures.sql]: this can't be right, merge conflicts were not properly solved

**Comment by BarbzYHOOL** [comment]
ok there was an accident, the PR got merged and I reverted it because it wasn't ready/tested

Can you open the same PR again @rbedfordpro ? (and sorry for that)
*also we'll lose the information posted here*

File: sql/world/Creatures.sql [hunk 2]
```diff
@@ -1,5 +1,4 @@
...
-(18649, 0, 0, 0, 0, 0, 18234, 0, 0, 0, 'Innkeeper Monica', NULL, NULL, 0, 1, 2, 0, 35, 65537, 0.8, 0.28571, 1, 0, 2, 2, 0, 26, 4.6, 2000, 1900, 1, 0, 2048, 0, 0, 0, 0, 0, 0, 1, 1, 100, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 1, 3, 1, 1, 1, 1, 0, 70, 1, 0, 2, 'npc_innkeeper', 12340),
+REPLACE INTO `creature_template` VALUES (70101, 0, 0, 0, 0, 0, 25901, 0, 0, 0, 'Xrispins', 'Guild House Seller', '', 0, 35, 35, 0, 35, 1, 1, 1.14286, 1, 0, 49, 64, 0, 118, 1, 2000, 2000, 1, 33536, 2048, 0, 0, 0, 0, 0, 0, 33, 49, 11, 7, 4096, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 0, 3, 1, 1, 1, 1, 0, 0, 1, 0, 0, 'GuildHouseSeller', 12340),
+(18649, 0, 0, 0, 0, 0, 18234, 0, 0, 0, 'Innkeeper Angela', NULL, NULL, 0, 1, 2, 0, 35, 65537, 0.8, 0.28571, 1, 0, 2, 2, 0, 26, 4.6, 2000, 1900, 1, 0, 2048, 0, 0, 0, 0, 0, 0, 1, 1, 100, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, '', 1, 3, 1, 1, 1, 1, 0, 70, 1, 0, 2, 'npc_innkeeper', 12340),
```
Comment (talamortis) [inline sql/world/Creatures.sql]: please do not change the name of the NPC as this is an ingame npc https://wotlk.evowow.com/?npc=18649

Comment (BarbzYHOOL) [reply sql/world/Creatures.sql]: Hmm good point, it should be an update statement in fact, and not a replace into !!! @rbedfordpro

Comment (rbedfordpro) [reply sql/world/Creatures.sql]: Good point! Thanks - have reverted the .sql for that Innkeeper Monica NPC

**Pull request merged**